pytorch_lightning/loops/batch/training_batch_loop.py

```diff
@@ -545,12 +550,16 @@ def _build_kwargs(self, batch: Any, batch_idx: int, opt_idx: int, hiddens: Optio
             the keyword arguments for the training step
         """
         # enable not needing to add opt_idx to training_step
-        step_kwargs = OrderedDict([("batch", batch), ("batch_idx", batch_idx)])
+        step_kwargs = OrderedDict({"batch": batch})
```

Comment:
Using list of tuples is important as dicts do not guarantee order in 3.6

Reply:
Done !